### ISSUE TITLE
Expose 24h report on the POS keypad for store owners

### ISSUE THREAD

**Issue by pavlenex** [btcpayserver/btcpayserver#6160]
We're noticing store owners really want to get number of sales + daily revenue, for them through the web pos, it's easiest to add UI in the quick transaction view, and expose some basic `today at a glance data` that could include number of sales, tips and total.

It's to be discussed if employee role should have access to this, perhaps only tips.

**Comment by pavlenex** [comment]
Hey @dafunction actually I was proposing we expose this in the keypad POS customer-facing view so that a cashier could quickly see the day at a glance without having to be in the backend. I wouldn't mind this in the dashboard either, looks quite cool, I'm sure @dstrukt and @dennisreimann can help with direction in terms of design.

**Comment by dafunction** [comment]
@pavlenex ahh ok.  is it this view, then?
![51144](https://github.com/user-attachments/assets/4fc3555e-5fa2-4549-8dae-38509f051f26)

in either case i'm happy to take a stab at implementing this with the support of @dstrukt and/or @dennisreimann.

**Comment by pavlenex** [comment]
Correct, my initial idea was to tuck it under "invoices" the left icon. I'll leave it up to @dstrukt to scope this one, but feel free to take a stab at it. There may be certain permissions involved wrt this one, this is where @dennisreimann can help with guidance.

**Comment by pavlenex** [comment]
@dstrukt This is relevant for both app and the server. I never said this should be in the right (product view), but the the left (invoice view). 

Merchant wants to be aware how much money they've made that day without an overhead of having to go look in the backoffice, which isn't an option when you have 20 hungry people in line, especially since we just share a public facing web-app link for merchants as logging them in is operational overhead.

The only problem here is how we handle this permission-wise, currently we do not allow logged out user to access these keypad views (transaction/product), tbh it's crazy a product view is disabled! 

Since all of this is on merchants device there needs to be an option for store owner to allow logged out user to preview product view or recent invoices and hopefully quick reporting.

There are two considerations here:

1. Merchant wants a quick way to preview their daily sales without an overhead
2. Do we want to allow logged out user to preview this (even as a store setting option)

I think the minimum we could do right now is ignore 2. and focus on 1. and figure out a way to persent quick at a glance reporting for merchants without them having to go back to the backoffice.


You're right, it's not easy thing to solve. Alternatively merchants could get an automated daily report by email, but this also has a complexity of setting this all up. 

We could leave it as a discussion for the next design call, and come back with the direction for @dafunction if we decide to proceed with this one.

**Comment by dafunction** [comment]
when are the design calls?  would love to join if possible if nothing else for context so i can be on the same page as everyone.  i'm in the Mattermost server under the same handle by the way.

**Comment by pavlenex** [comment]
@rockstardev I remember we talked about problem 2) **How do we log people in through a QR of POS.** I think I have an idea here. 

How about that in the App QR that we already have we add a switch (toggle) or a drop down, where store owner can select the user they'd like to log in, and QR then refreshes? I am unsure if it's a good approach vs having it as a store setting, but I think it's a more flexible one. If you think it's useful, we could log this as a separate issue.

**Comment by pavlenex** [comment]
> when are the design calls? would love to join if possible if nothing else for context so i can be on the same page as everyone. i'm in the Mattermost server under the same handle by the way.

There's UX/UI channel, ping me there and I'll send you an invite for the next call.
